data_training.ipynb

```diff
@@ -0,0 +1,305 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5fc323c2-3566-404c-8ef1-ae6084e1b28f",
+   "metadata": {},
+   "source": [
+    "### Feature Scaling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a278e736-596e-4fe5-9c1e-1c5816387a0e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "numerical_features = [\"Age\", \"Height\", \"Weight\", \"FCVC\", \"NCP\", \"CH2O\", \"FAF\", \"TUE\"]\n",
+    "\n",
+    "# Fit the scaler to the train set, not the test set, to prevent \"leaking\" of the test set.\n",
+    "scaler = StandardScaler().fit(X_train[numerical_features])\n",
+    "\n",
+    "X_train[numerical_features] = scaler.transform(X_train[numerical_features])\n",
+    "X_test[numerical_features] = scaler.transform(X_test[numerical_features])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "105a6d38-a058-4cc0-9c55-655adf9da64b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42669e1c-bb68-40d6-8729-f99fc24ada22",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41531b10-a447-44f0-b6a6-e0ffcfd47737",
+   "metadata": {},
+   "source": [
+    "### Categorical Feature Encoding"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4187423-a69e-46c3-a329-cb5ff18cf639",
+   "metadata": {},
+   "source": [
+    "#### Target Feature"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4866270c-96df-429a-bdcb-bf34028bd4cf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "le = LabelEncoder()\n",
+    "le.fit(ob_df[\"NObeyesdad\"])\n",
+    "\n",
+    "y_train[:] = le.transform(y_train)\n",
+    "y_test[:] = le.transform(y_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6e11f195-e3af-42a7-b40a-7681993a87e0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8dcde8a8-0cb5-4d4a-8d85-8203da1f69f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4797f44a-06ef-46ec-84d8-1f434d8c988f",
+   "metadata": {},
+   "source": [
+    "#### Non-Target Features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2286adf5-6de0-493e-86cc-41f4dbd32d9d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nominal_features = [\n",
+    "    \"Gender\",\n",
+    "    \"family_history_with_overweight\",\n",
+    "    \"FAVC\",\n",
+    "    \"CAEC\",\n",
+    "    \"SMOKE\",\n",
+    "    \"SCC\",\n",
+    "    \"CALC\",\n",
+    "    \"MTRANS\"\n",
+    "]\n",
+    "\n",
+    "ohe = OneHotEncoder(sparse_output=False)\n",
```

Comment:
May want to specify `drop="first"`. I'll need to read more about it.

Reply:
It's not so straightforward. It depends on the model we're going to use and whether regularization is applied. If regularization is used, then we shouldn't drop the first column. For now, I'm going to drop the first column — we can revisit this later.